### PR TITLE
A4A: Creating a contact support hash linked to the contact form button, allowing it to open when using the URL

### PR DESCRIPTION
**Pull request by coder-karen**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/304

## Proposed Changes

* This PR adds a #contact-support hash, and makes sure that it is set when opening the modal. It also then checks to see if that hash is added to the URL and will open the contact support modal if so.

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/overview`.

* On the overview page, click on the Contact Support button. The support form modal should open, and the URL should have `#contact-support` appended to it. Click to close the modal - the modal should close and `#contact-support` removed from the URL.
* Visit the URL directly, in a new tab, for example `http://agencies.localhost:3000/overview#contact-support`. The modal should open immediately.
* Check that the `calypso_a4a_overview_contact_support_click` event fires once in each case, when clicking the Contact Support button.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/overview/sidebar/contact-support/index.tsx [hunk 1]
```diff
@@ -7,16 +7,31 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
...
 
-	const onContactSupport = useCallback( () => {
-		dispatch( recordTracksEvent( 'calypso_a4a_overview_contact_support_click' ) );
-		setShowUserSupportForm( true );
-	}, [ dispatch ] );
+	const toggleContactForm = () => {
+		setShowUserSupportForm( ( prevState ) => {
+			const nextState = ! prevState;
+
+			const hashFragment = nextState ? CONTACT_URL_HASH_FRAGMENT : '';
```
Comment (gogdzl) [inline client/a8c-for-agencies/sections/overview/sidebar/contact-support/index.tsx:24]: We could rename this var to contactUrlHashFragment, what do you think?

**Review by gogdzl — APPROVED** [review]
Looks good to me, the only drawback is that it triggers a page refresh when the user clicks the 'Back' button in their browser, but I think that supporting this is unneeded, since it is going back to the right place.

**Comment by coder-karen** [comment]
@gogdzl Thanks for reviewing! That's a good point, though it looks like that's the case on trunk too so not related to this PR. I notice I don't go back to Overview though, I go back to the previous page if my only click on Overview is the contact support button. However if I navigate to the contact support link directly, clicking back takes me to whatever site I was on previously. So, I'm not sure if you are experiencing something different?

File: client/a8c-for-agencies/sections/overview/sidebar/contact-support/index.tsx (hunk 1 above)
Comment (coder-karen) [reply client/a8c-for-agencies/sections/overview/sidebar/contact-support/index.tsx:24]: All our variable contants take that 'screaming' snake case format - those that refer to strings anyway, from what I've seen (eg `wp-calypso/client/a8c-for-agencies/sections/sites/constants.ts`, `wp-calypso/client/a8c-for-agencies/sections/sites/features/features.ts`). 
It looks like the naming conventions here support that: `wp-calypso/docs/coding-guidelines/javascript.md` - for 'Variables intended to be used as a [constant]' - so really that would include any constant value (string, int..)

The idea is that that const can be exported and used elsewhere if needed too - perhaps we want to add the link somewhere else, so it's not necessarily just limited to that file.

**Comment by gogdzl** [comment]
> though it looks like that's the case on trunk too so not related to this PR.

Tested again and you are right! I think I missed a Calypso rebuild when testing.

**Pull request merged**